src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2765,6 +2766,45 @@ private bool IsMisplacedModifier(SyntaxListBuilder modifiers, SyntaxList<Attribu
             return false;
         }
 
+        private static bool ContainsErrorDiagnostic(GreenNode node)
+        {
+            // ContainsDiagnostics returns true if this node (or any descendents) contain any sort of error.  However,
+            // GetDiagnostics() only returns diagnostics at that node itself.  So we have to explicitly walk down the
+            // tree to find out if the diagnostics are error or not.
+
+            // Quick check to avoid any unnecessary work.
+            if (node.ContainsDiagnostics)
+            {
+                var stack = ArrayBuilder<GreenNode>.GetInstance();
```

Comment:
Would `node.DescendantNodesAndTokens().Any(static n => n.GetDiagnostics().Any(static d => d.Severity == DiagnosticSeverity.Error))` or some such do the trick?

Reply:
this is a greennode, so no DescendantNodesAndTokens.  One of the downsides of working at the green layer (as well as GetDiagnostics itself not being recursive).